codalab/apps/web/static/js/worksheet/worksheet_side_panel.js

```diff
@@ -234,11 +234,17 @@ var BundleDetailSidePanel = React.createClass({
     },
 
     updateFileBrowser: function(folder_path) {
-        if (folder_path == '..')  // Go to parent directory
-          folder_path = this.state.currentWorkingDirectory.replace(/\/[^\/]*$/, '');
-
-        if (this.state.currentWorkingDirectory != '')
-            folder_path = this.state.currentWorkingDirectory + "/" + folder_path;
+        if (folder_path == '..'){  // Go to parent directory
+            folder_path = this.state.currentWorkingDirectory.substring(0, this.state.currentWorkingDirectory.lastIndexOf('/'));
+        }
+        else if (this.state.currentWorkingDirectory != '') {
+            if (folder_path != ''){
+                folder_path = this.state.currentWorkingDirectory + "/" + folder_path;
+            }
+            else {
+                folder_path = this.state.currentWorkingDirectory;
+            }
+        }
```

Comment:
This exactly happens on the root of the worksheet. The files and folders present at that folder are returned in that case.